R/pkg/R/sparkR.R

```diff
@@ -558,16 +558,18 @@ sparkCheckInstall <- function(sparkHome, master) {
       message(msg)
       NULL
     } else {
-      if (!nzchar(master) || isMasterLocal(master)) {
+      if (isMasterLocal(master)) {
         msg <- paste0("Spark not found in SPARK_HOME: ",
                       sparkHome)
         message(msg)
         packageLocalDir <- install.spark()
         packageLocalDir
-      } else {
+      } else if (master == "yarn-client") {
         msg <- paste0("Spark not found in SPARK_HOME: ",
                       sparkHome, "\n", installInstruction("remote"))
         stop(msg)
+      } else {
```

Comment:
also what happens to "mesos"? this is what we are saying we should support in SparkR 2.1.0